File: python/misarblog/embed.py
from urllib.parse import urlencode

EMBED_BASE = "https://misar.blog"


def embed_url(username: str, slug: str | None = None, theme: str = "auto") -> str:
    path = f"{username}/{slug}/embed" if slug else f"{username}/embed"
    url = f"{EMBED_BASE}/{path}"
    if theme != "auto":
        url = f"{url}?{urlencode({'theme': theme})}"
    return url

File: ruby/lib/misarblog/embed.rb
module MisarBlog
  EMBED_BASE = "https://misar.blog".freeze

  def self.embed_url(username:, slug: nil, theme: "auto")
    url = "#{EMBED_BASE}/#{username}"
    url = "#{url}/#{slug}" if slug && !slug.empty?
    url = "#{url}/embed"
    url = "#{url}?theme=#{theme}" if theme != "auto"
    url
  end
end
